kpops/components/base_components/kubernetes_app.py

```diff
@@ -88,8 +96,9 @@ class Config(CamelCaseConfig, DescConfig):
         pass
 
     def __init__(self, **kwargs):
+        if kwargs.get("validate_name", True):
+            self.validate_kubernetes_name(kwargs["name"])
```

Comment:
A component is initialized more than once before reaching its "final" form.
[first](https://github.com/bakdata/kpops/pull/198/files#diff-bf5691f6b8f231bf35e494f9dbb7d78f8bff5467239c2a3f9a93af4a78e68bd1R185-R190)
[second](https://github.com/bakdata/kpops/pull/198/files#diff-bf5691f6b8f231bf35e494f9dbb7d78f8bff5467239c2a3f9a93af4a78e68bd1R243-R244)

Since substitution happens after at least one init and right before the last one, it is possible that the `name` of a `KubernetesApp` instance contains placeholders. Hence, we can only validate the name during the final init after the substitution is carried out.